client-runtime/src/main/java/com/microsoft/rest/http/RxNettyAdapter.java

```diff
@@ -62,6 +63,11 @@ private SSLEngine getSSLEngine(String host) {
     }
 
     @Override
+    public HttpClient withProxy(Proxy proxy) {
```

Comment:
It looks like this is starting down a road of allowing mutation of an HttpClient instance after it's created, which gives me pause. If you find it's significantly easier to work with than an immutable class/class hierarchy, I can deal with it.

Reply:
Basically, I aspired to make RestClient and its contents immutable so that you can share it with whatever other services impls or clients (e.g. RefreshTokenClient) that need it to configure itself without concern that it might mess up the configuration of the original service.

Reply:
This is a really good point. I'll change this to go on the Builder instead of the RestClient.

Reply:
Scratch that. I'll make this go on the constructor of our HttpClients.

Reply:
Scratch that. How an HttpClient uses a Proxy is pretty implementation specific, so I created an RxNettyAdapter Builder that has a ```withProxy(Proxy)``` method on it.